engines/cryo/eden.cpp

```diff
@@ -4729,29 +4729,30 @@ void EdenGame::convertMacToPC() {
 }
 
 void EdenGame::loadpermfiles() {
-	switch (_vm->getPlatform()) {
-	case Common::kPlatformDOS:
-	{
-		// Since PC version stores hotspots and rooms info in the executable, load them from premade resource file
-		Common::File f;
+	Common::File f;
+	const int kNumIcons = 136;
+	const int kNumRooms = 424;
 
-		if (f.open("cryo.dat")) {
-			const int kNumIcons = 136;
-			const int kNumRooms = 424;
-			const int dataSize = f.size() - 8 - 1;	// CRYODATA + version
-			char headerId[9];
-			
-			f.read(headerId, 8);
-			headerId[8] = '\0';
-			if (strcmp(headerId, "CRYODATA"))
-				error("Invalid aux data file");
+	if (f.open("cryo.dat")) {
+		const int dataSize = f.size() - 8 - 1;	// CRYODATA + version
+		char headerId[9];
 
-			if (f.readByte() != CRYO_DAT_VER)
-				error("Incorrect aux data version");
+		f.read(headerId, 8);
+		headerId[8] = '\0';
+		if (strcmp(headerId, "CRYODATA"))
+			error("Invalid aux data file");
 
-			if (dataSize != kNumIcons * sizeof(Icon) + kNumRooms * sizeof(Room))
-				error("Mismatching data in aux data file");
+		if (f.readByte() != CRYO_DAT_VER)
+			error("Incorrect aux data version");
 
+		if (dataSize != kNumIcons * sizeof(Icon) + kNumRooms * sizeof(Room))
```

Comment:
This check is very sketchy and may be incompatible across different platforms because sizeof(Icon) and Room may vary, even though they are declared as a byte-packed structures currently. It's probably better to use fixed constants here.